lib/public/components/Filters/RunsFilter/durationFilter.js

```diff
@@ -11,26 +11,68 @@
  * or submit itself to any jurisdiction.
  */
 
-import { amountFilter } from '../common/filters/amountFilter.js';
+import { h } from '/js/src/index.js';
+import { comparisonOperatorFilter } from '../common/filters/comparisonOperatorFilter.js';
 
 /**
  * Returns the run duration filter component
  *
- * @param {RunsOverviewModel} runModel the runs model object
- *
- * @return {vnode} the duration filter
+ * @param {DurationInputModel} durationFilterModel the duration input model
+ * @param {ComparisonOperatorSelectionModel} operatorSelectionModel the comparison operator selection model
+ * @return {Component} the duration filter
  */
-export const durationFilter = (runModel) => amountFilter(
-    runModel.runDurationFilter,
-    (filter) => {
-        runModel.runDurationFilter = filter;
-    },
-    {
-        operatorAttributes: {
-            id: 'duration-operator',
+export const durationFilter = (durationFilterModel, operatorSelectionModel) => {
+    const { hours, minutes, seconds } = durationFilterModel.raw;
+
+    const hoursInput = h('input.flex-grow', {
+        id: 'hours-input',
+        type: 'number',
+        min: 0,
+        value: hours,
+        oninput: (e) => {
+            const { value } = e.target;
+            durationFilterModel.update({ hours: value.length === 0 ? null : Number(value) });
+        },
+    });
+    const minutesInput = h('input.flex-grow', {
+        id: 'minutes-input',
+        type: 'number',
+        min: 0,
+        max: 59,
+        value: minutes,
+        oninput: (e) => {
+            const { value } = e.target;
+            durationFilterModel.update({ minutes: value.length === 0 ? null : Number(value) });
         },
-        limitAttributes: {
-            id: 'duration-limit',
+    }, 'm');
+    const secondsInput = h('input.flex-grow', {
+        id: 'seconds-input',
+        type: 'number',
+        min: 0,
+        max: 59,
+        value: seconds,
+        oninput: (e) => {
+            const { value } = e.target;
+            durationFilterModel.update({ seconds: value.length === 0 ? null : Number(value) });
         },
-    },
-);
+    }, 's');
+
+    const inputs = h('.flex-row.w-40', [
```

Comment:
You should not apply things like `w-40` here, because then you can't properly reuse this input anywhere.

Reply:
applied